flow/shared/worklow.go

```diff
@@ -0,0 +1,28 @@
+package shared
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"go.temporal.io/sdk/client"
+
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+)
+
+func GetWorkflowStatus(ctx context.Context, temporalClient client.Client, workflowID string) (protos.FlowStatus, error) {
+	res, err := temporalClient.QueryWorkflow(ctx, workflowID, "", FlowStatusQuery)
+	if err != nil {
+		slog.Error(fmt.Sprintf("failed to get status in workflow with ID %s: %s", workflowID, err.Error()))
+		return protos.FlowStatus_STATUS_UNKNOWN,
+			fmt.Errorf("failed to get status in workflow with ID %s: %w", workflowID, err)
```

Comment:
```suggestion
		slog.Error(fmt.Sprintf("failed to query status in workflow with ID %s: %s", workflowID, err.Error()))
		return protos.FlowStatus_STATUS_UNKNOWN,
			fmt.Errorf("failed to query status in workflow with ID %s: %w", workflowID, err)
```
makes error messages distinct so that when looking up error messages in code we know which line was hit